tests/test_main.py

```diff
@@ -304,7 +312,12 @@ def test_packages_distributions_example2(self):
 
 
 class PackagesDistributionsTest(
-    fixtures.OnSysPath, fixtures.SiteDir, unittest.TestCase
+    fixtures.EggInfoPkg,
+    fixtures.EggInfoPkgPipInstalledNoModules,
+    fixtures.EggInfoPkgSourcesFallback,
```

Comment:
These fixtures aren't needed by other tests in this class. Perhaps it would be better to have a separate class or mechanism to inject these behaviors for the tests that rely on them.

Reply:
Ack. I have to admit I'm not too familiar with how these fixtures interact with `unittest.TestCase`-style tests, and how I would proceed to apply the fixtures to one test method only.

Will instead move the method to a new `PackagesDistributionsEggTest` class.

Reply:
Done in [b391f77](https://github.com/python/importlib_metadata/pull/437/commits/b391f77d078b461da972226dbf6f0649e78eb5db)